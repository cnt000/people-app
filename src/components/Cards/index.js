import React from "react"
import PropTypes from 'prop-types'
import styled, { keyframes } from "styled-components"
import Card from "../Card"

const gradient = keyframes`
  0% {
    background-position: 0% 50%
  }
  50% {
    background-position: 100% 50%
  }
  100% {
    background-position: 0% 50%
  }
`

const Div = styled.div`
  width: 90vw;
  height: 90vh;
  border: 1px solid black;
  background: linear-gradient(-45deg, #EE7752, #E73C7E, #23A6D5, #23D5AB);
  background-size: 170% 100%;
  animation: ${gradient} 15s ease infinite;
  border-radius: 0.25em;
  padding: .4em;
`

const Ul = styled.ul`
  display: grid;
  grid-template-columns: 25% 25% 25% 25%;
  grid-template-rows: 25% 25% 25% 25%;
  margin: 0;
  padding: 0;
`
const Li = styled.li`
  list-style: none;
  margin: 0;
  padding: 0;
`

const Cards = ({ cards, showCard }) => (
  <Div>
    <Ul>
      {cards && cards.map((card, position) =>
        <Li key={position}>
          <Card
            value={card.value}
            showed={card.showed}
            onClick={() => showCard(position)}
          />
        </Li>
      )}
    </Ul>
  </Div>
)

Cards.propTypes = {
  cards: PropTypes.arrayOf(
    PropTypes.shape({
      value: PropTypes.string.isRequired,
      showed: PropTypes.bool.isRequired
    }).isRequired
  ).isRequired
}

export default Cards
