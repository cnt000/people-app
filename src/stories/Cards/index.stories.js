import React from 'react';

import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import Cards from "../../components/Cards"
import defaultState from "../../defaultState"

const cards_two = [
  {
    value: "A",
    showed: false
  },
  {
    value: "A",
    showed: false
  }
]

const cards_four = [
  {
    value: "C",
    showed: false
  },
  {
    value: "D",
    showed: false
  },
  {
    value: "C",
    showed: false
  },
  {
    value: "D",
    showed: false
  }
]

const cards_six = [
  {
    value: "A",
    showed: false
  },
  {
    value: "B",
    showed: false
  },
  {
    value: "C",
    showed: false
  },
  {
    value: "A",
    showed: false
  },
  {
    value: "B",
    showed: false
  },
  {
    value: "C",
    showed: false
  }
]

storiesOf('Cards', module)
  .add('default state', () => <Cards cards={defaultState.cards} showCard={action('show')} />)
  .add('2 cards', () => <Cards cards={cards_two} showCard={action('show')} />)
  .add('4 cards', () => <Cards cards={cards_four} showCard={action('show')} />)
  .add('6 cards', () => <Cards cards={cards_six} showCard={action('show')} />)
  .add('empty', () => <Cards cards={[]} showCard={action('show')} />)
