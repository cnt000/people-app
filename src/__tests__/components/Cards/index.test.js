import React from 'react'
import ReactDOM from 'react-dom'
import Cards from '../../../components/Cards'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import defaulState from '../../../defaultState'

describe('Cards', () => {
  it('should render (without snapshot)', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Cards cards={defaulState.cards} />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  // it('should render a <ul>', () => {
  //   const wrapper = shallow(<Cards cards={[{
  //     value: "DDD",
  //     showed: false
  //   }]} />);
  //   const value = <ul></ul>;
  //   expect(wrapper.contains(value)).toEqual(true);
  // });

  it('should render correctly empty', () => {
    const output = shallow(<Cards cards={[]} />)
    expect(shallowToJson(output)).toMatchSnapshot()
  })

  it('should render correctly with snapshot', () => {
    const output = shallow(<Cards cards={defaulState.cards} />)
    expect(shallowToJson(output)).toMatchSnapshot()
  })
})
