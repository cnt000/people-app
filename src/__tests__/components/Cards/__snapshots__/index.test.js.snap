// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards should render correctly empty 1`] = `
<styled.div>
  <styled.ul />
</styled.div>
`;

exports[`Cards should render correctly with snapshot 1`] = `
<styled.div>
  <styled.ul>
    <styled.li
      key="0"
    >
      <Card
        onClick={[Function]}
        showed={false}
        value="AAA"
      />
    </styled.li>
    <styled.li
      key="1"
    >
      <Card
        onClick={[Function]}
        showed={false}
        value="BBB"
      />
    </styled.li>
    <styled.li
      key="2"
    >
      <Card
        onClick={[Function]}
        showed={false}
        value="CCC"
      />
    </styled.li>
    <styled.li
      key="3"
    >
      <Card
        onClick={[Function]}
        showed={false}
        value="DDD"
      />
    </styled.li>
    <styled.li
      key="4"
    >
      <Card
        onClick={[Function]}
        showed={false}
        value="AAA"
      />
    </styled.li>
    <styled.li
      key="5"
    >
      <Card
        onClick={[Function]}
        showed={false}
        value="BBB"
      />
    </styled.li>
    <styled.li
      key="6"
    >
      <Card
        onClick={[Function]}
        showed={false}
        value="CCC"
      />
    </styled.li>
    <styled.li
      key="7"
    >
      <Card
        onClick={[Function]}
        showed={false}
        value="DDD"
      />
    </styled.li>
  </styled.ul>
</styled.div>
`;
